sql/core/src/main/scala/org/apache/spark/sql/api/java/UDFRegistration.scala

```diff
@@ -94,159 +93,159 @@ private[java] trait UDFRegistration {
   */
 
   // scalastyle:off
-  def registerFunction(name: String, f: UDF1[_, _], dataType: DataType) = {
-    val scalaType = asScalaDataType(dataType)
+  def registerFunction(
+      name: String, f: UDF1[_, _], @transient dataType: DataType) = {
     sqlContext.functionRegistry.registerFunction(
       name,
-      (e: Seq[Expression]) => ScalaUdf(f.asInstanceOf[UDF1[Any, Any]].call(_: Any), scalaType, e))
+      (e: Seq[Expression]) => ScalaUdf(f.asInstanceOf[UDF1[Any, Any]].call(_: Any), dataType, e))
```

Comment:
Not in this PR, but we should add `TypeTags` to these functions for the input arguments.